warp10/src/main/java/io/warp10/script/functions/TOJSON.java

```diff
@@ -43,120 +37,22 @@ public TOJSON(String name) {
   public Object apply(WarpScriptStack stack) throws WarpScriptException {
     Object o = stack.pop();
 
-    JsonSerializer parser = BOON_SERIALIZER_FACTORY.create();
-
     //
     // Only allow the serialization of simple lists and maps, otherwise JSON might
     // expose internals
     //
 
-    if (!validate(o)) {
-      throw new WarpScriptException(getName() + " can only serialize structures containing numbers, strings, booleans, lists and maps which do not reference the same list/map multiple times.");
+    try {
+      Long maxJsonSize = (Long)stack.getAttribute(WarpScriptStack.ATTRIBUTE_JSON_MAXSIZE);
+      String json = JsonUtils.objectToJson(o, false, maxJsonSize);
```

Comment:
This call will fail with an NPE if the stack attribute retrieved above is not set. It is set in the case of MemoryWarpScriptStack instances but potentially not in other implementations of WarpScriptStack.

Same in MAXJSON

Reply:
I think this is the case for all other limits.
It this is not set, what should be the value? The default one or the max one?

Reply:
Better fix that in another PR as all other limit attributes suffer from the same problem.